pennylane/tape/tape.py

```diff
@@ -94,6 +94,9 @@ def expand_tape(qscript, depth=1, stop_at=None, expand_measurements=False):
     Returns:
         QuantumScript: The expanded version of ``qscript``.
 
+    Raises:
+        QuantumFunctionError: if some observables in the qscript are not qubit-wise commuting
```

Comment:
Is this raised from a separate function, or are there validation checks in here that raise the error manually? If the former, then I don't think we need it in this doc-string.

Reply:
it's raised here manually :) non-qwc observables will raise a TypeError when trying to diagonalize, so we catch it and raise this error instead